Microsoft/ApplicationInsights-Java
# Major optimization to smokeTests (let's merge this before other smokeTest changes)

This reduces the number of containers running at a time (now only one at a time) and increased perfomance by ~2x: before 24 tests in ~19.1 minutes, now they run in ~9.1 min.

This came about because I found that when I added new appservers, things became very slow. Often, the current flow crashes docker and requires manual intervention. This was mostly because at the end of a test class, all the containers used would still be open and trying to stop in parallel. Now only one container is active at a time and it life spans the "parameter vector scope" instead of "test method scope".

Currently the smoketests workflow is like this:
For each test class (6 test classes):
1. Compute parameters (determine which containers should be run)
2. Select a param vector
3. start container
4. run one test method
5. if more params, goto 2
6. stop all containers

Now:
1. Compute params
2. Select param vector
3. start container
4. run all test methods
5. stop container
6. if more params, goto 2
